pkg/splunk/enterprise/monitoringconsole.go

```diff
@@ -313,16 +367,21 @@ func AddURLsConfigMap(revised *corev1.ConfigMap, crName string, newURLs []corev1
 }
 
 //DeleteURLsConfigMap for deleting server peers to the monitoring console or scaling down
-func DeleteURLsConfigMap(revised *corev1.ConfigMap, crName string, newURLs []corev1.EnvVar, deleteCR bool) {
+func DeleteURLsConfigMap(revised *corev1.ConfigMap, crName string, newURLs []corev1.EnvVar, deleteCR bool) []string {
+	var deletedPeers []string
 	for _, url := range newURLs {
 		currentURLs := strings.Split(revised.Data[url.Name], ",")
 		sort.Strings(currentURLs)
 		for _, curr := range currentURLs {
 			//scale DOWN
 			if strings.Contains(curr, crName) && !strings.Contains(url.Value, curr) && !deleteCR {
 				revised.Data[url.Name] = strings.ReplaceAll(revised.Data[url.Name], curr, "")
+				deletedPeers = append(deletedPeers, curr)
 			} else if strings.Contains(curr, crName) && deleteCR {
 				revised.Data[url.Name] = strings.ReplaceAll(revised.Data[url.Name], url.Value, "")
+				if url.Name != "SPLUNK_MULTISITE_MASTER" {
+					deletedPeers = strings.Split(url.Value, ",")
```

Comment:
Should we append here? Or directly assign?

Reply:
url.Value is the entire string here with multiple URLs example "a,b,c,c" and we need []string = {"a", "b", "c"}

Reply:
Directly assigning since either we will scale down or delete cr. So in case of scaling down, we are appending one by one and when deleting entire cr we add all the instances names that need to be deleted